proposals/0037-cbuffer-contexts.md

```diff
@@ -69,39 +67,62 @@ To simplify HLSL's language semantics and the compiler implementation a new
 grammar formation is adopted for cbuffers:
 
 ```latex
-
 \begin{grammar}
-  \define{cbuffer-declaration-group}\br
-  \terminal{cbuffer} identifier \opt{resource-binding} \terminal{\{}
-  \opt{cbuffer-declaration-seq} \terminal {\}}\br
-
-  \define{cbuffer-declaration-seq}\br
-  cbuffer-declaration\br
-  cbuffer-declaration-seq cbuffer-declaration\br
-
   \define{cbuffer-declaration}\br
-  variable-declaration\br
-  empty-declaration\br
+  \terminal{cbuffer} name \opt{resource-binding} \terminal{\{}
+    \opt{cbuffer-member-seq} \terminal {\}}\br
+
+  \define{cbuffer-member-seq}\br
+  cbuffer-member-declaration\br
+  cbuffer-member-seq cbuffer-member-declaration\br
+
+  \define{cbuffer-member-declaration}\br
+  block-declaration\br
+  function-definition\br
+  template-declaration\br
+  empty-declaration
```

Comment:
Where did the `variable-declaration` disappear? And what is `block-declaration`?

Reply:
`block-declaration` is any declaration that can occur at block scope (i.e. any declaration you could have inside a function). That includes variables, but also aliases, static asserts, and other "simple" declarations like functions and classes.

The draft language spec PR includes some wider changes to the declarations chapter and grammars:
https://github.com/microsoft/hlsl-specs/pull/684/files#diff-11b83f8183acd1a33489925bf65fbeb9d420609588c72b0075263be859f6d186

Reply:
Oh I see :) That's why I could not find it in the current version of the spec.